vector/src/main/java/im/vector/analytics/Events.kt

```diff
@@ -24,10 +28,16 @@ enum class Action(val value: String) {
  * @param title the title associated with the event
  * @param value the optional value associated with the event
  */
-sealed class Event(val category: Category, val action: Action, val title: String? = null, val value: Float? = null) {
-    data class InitialSync(val duration: Long) : Event(Category.METRICS, Action.STARTUP, "initialSync", duration.toFloat())
-    data class IncrementalSync(val duration: Long) : Event(Category.METRICS, Action.STARTUP, "incrementalSync", duration.toFloat())
-    data class StorePreload(val duration: Long) : Event(Category.METRICS, Action.STARTUP, "storePreload", duration.toFloat())
-    data class LaunchScreen(val duration: Long) : Event(Category.METRICS, Action.STARTUP, "launchScreen", duration.toFloat())
-    data class Rooms(val nbOfRooms: Int) : Event(Category.METRICS, Action.STATS, "rooms", nbOfRooms.toFloat())
+sealed class TrackingEvent(val category: Category, val action: Action, val title: String? = null, val value: Float? = null) {
```

Comment:
it seems that default value are never used and parameters are never null. Simplify signature?

Reply:
The title and value are optional for majority of Analytics solutions, including Piwik. It seems legit to keep that in mind for Events.

Reply:
ok